src/pudl/metadata/fields.py

```diff
@@ -2143,8 +2186,10 @@
 
 def get_pudl_dtypes(
     group: str | None = None,
+    resource: str | None = None,
```

Comment:
Would this resource specific selection still be necessary if the PPE were listed as part of the EIA data source (which wouldn't seem crazy). Or I guess it's the need to override with the resource specific Categorical types.

Did you look at the `Resource.format_df()` method as an alternative by chance?

Reply:
I might have confused myself with the different metadata classes and functionality but walking through my thought process:
- `RESOURCE_METADATA` is the metadata for resources in the DB, so the PPE shouldn't be added as a resource here since it's just an output table. 
- I could use `Resource.format_df()` by defining a one off PPE resource and not add it to `RESOURCE_METADATA` but I think the only functionality I'd really be leveraging from `Resource.format_df()` is to set specific column types (in this case `category`), so I might as well use `df.astype()` to do that.
- In the future these fields will be `category` types as per #1866 and `apply_pudl_dtypes()` can be used without any group or resource overrides.
- For now, maybe these `category` type overrides should just happen in a helper function in the PPE module?
- Does it make sense to have any of the PPE fields be included in the `FIELD_METADATA` or should all the specific type conversion just happen in a PPE module helper function?

Reply:
The metadata for the DB is defined by a `Package` which contains a list of `Resource` objects. So you could define`Resource` objects for the output tables, composed of fields that are defined in `fields.py` and just make sure that the output table `Resource`s aren't included in the DB definition.